internal/spec/spec.go

```diff
@@ -0,0 +1,90 @@
+package spec
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/Microsoft/hcsshim/internal/guestpath"
+)
+
+// DevShmMountWithSize returns a /dev/shm device mount with size set to
+// `sizeString` if it represents a valid size in KB, returns error otherwise.
+func DevShmMountWithSize(sizeString string) (*specs.Mount, error) {
+	if runtime.GOARCH == "windows" {
+		return nil, errors.New("only linux runtimes are supported")
+	}
+
+	size, err := strconv.ParseInt(sizeString, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("/dev/shm size must be a valid integer: %w", err)
+	}
+	if size <= 0 {
+		return nil, fmt.Errorf("/dev/shm size must be a positive integer, got: %d", size)
+	}
+
+	// Use the same options as in upstream https://github.com/containerd/containerd/blob/0def98e462706286e6eaeff4a90be22fda75e761/oci/mounts.go#L49
+	sizeKB := fmt.Sprintf("size=%sk", sizeString)
+	return &specs.Mount{
+		Source:      "shm",
+		Destination: "/dev/shm",
+		Type:        "tmpfs",
+		Options:     []string{"nosuid", "noexec", "nodev", "mode=1777", sizeKB},
+	}, nil
+}
+
+// GenerateWorkloadContainerNetworkMounts generates an array of specs.Mount
+// required for container networking. Original spec is left untouched and
+// it's the responsibility of a caller to update it.
+func GenerateWorkloadContainerNetworkMounts(sandboxID string, spec *specs.Spec) []specs.Mount {
```

Comment:
should we also return an error for windows runtimes here?

Reply:
reworking this to be Linux only.